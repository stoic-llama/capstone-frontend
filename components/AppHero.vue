<template>
    <div class="greypicture">
        <section class="masthead" role="img" aria-label="Image Description">
            <h1>Find Infant Formula</h1>
            <AppSearch />
        </section>
    </div>
</template>

<script>

</script>

<style lang="scss" scoped>
// Refer https://css-tricks.com/apply-a-filter-to-a-background-image/ for grey background
// Refer https://hero-generator.netlify.app for generating boilerplate
.greypicture {
  position: relative;
}
.greypicture::before {
  content: "";
  position: absolute;
  top: 0; left: 0;
  width: 100%; 
  height: 100vh;
  background-image: url(~/static/Formula_shortage.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  opacity: 0.5;
  filter: grayscale(100%);
}
.masthead {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  text-align: center;
  width: 100%;
  height: 100vh; /* if you don't want it to take up the full screen, reduce this number */
  overflow: hidden;
  background-size: cover !important;
  background: radial-gradient(ellipse at center, rgba(0,0,0,0) 0%, rgba(0,0,0,0) 0%, rgba(0,0,0,0.95) 100%), no-repeat center center scroll;
  // background: radial-gradient(ellipse at center, rgba(0,0,0,0) 0%, rgba(0,0,0,0) 37%, rgba(0,0,0,0.85) 100%), url(~/static/Formula_shortage.jpg) no-repeat center center scroll;
}
h1 {
  font-style: normal;
  font-weight: bold;
  color: #ffbe0b;
  font-size: 7vmin;
  letter-spacing: 0.03em;
  line-height: 1;
  text-shadow: 1px 2px 4px rgba(0, 0, 0, 0.8);
  margin-bottom: 40px;
}           
</style>