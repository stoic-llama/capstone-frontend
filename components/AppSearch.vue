<template>
    <form class="input-group input-group-lg" @submit.prevent="submitQuery(query)">
        <span class="input-group-text">
            <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-search" viewBox="0 0 16 16">
            <path d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"/>
            </svg>            
        </span>
        <input type="text" class="form-control" v-model="query" placeholder="Search by infant formula name..." aria-label="Baby Formula Name">
        <button class="btn btn-dark round" type="button" id="button-addon1" @click="submitQuery(query)">Search</button>
    </form>
</template>

<script>
export default {
name: 'AppSearch',
data() {
    return {
        query: ''
    }
},
methods: {
    submitQuery(q) {
        // this.$store.commit('UPDATE_QUERY', q)
        this.$store.dispatch('actionUpdateQuery', this.query)
        .then(() => {
            this.$router.push('/product/catalog')
        })
    }
}
};
</script>

<style lang="scss" scoped>
.input-group {
max-width: 584px;
margin-bottom: 10rem;
}
.input-group-text {
background-color: white;
border: 0;
border-radius: 1.5rem;
}
.form-control {
border-left-width: 0;
border-color: white;
}
.round {
border: 0;
border-radius: 1.5rem;
min-width: 6.5rem;
color: #ffbe0b;
font-weight: 100;
font-size: 1rem;
}
::placeholder {
font-size: 1rem;
}
@media screen and (max-width: 585px) {
.input-group {
    max-width: 400px;
}
}
</style>