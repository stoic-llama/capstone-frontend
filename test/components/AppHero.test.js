import { mount } from '@vue/test-utils'
import AppHero from '@/components/AppHero.vue'
import AppSearch from '@/components/AppSearch.vue'

test('mount AppHero component', () => {
    const wrapper = mount(AppHero)

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findComponent(AppSearch).exists()).toBe(true);
})


