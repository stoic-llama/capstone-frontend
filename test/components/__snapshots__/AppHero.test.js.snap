// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount AppHero component 1`] = `
"<div class="greypicture">
  <section role="img" aria-label="Image Description" class="masthead">
    <h1>Find Infant Formula</h1>
    <form class="input-group input-group-lg"><span class="input-group-text"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" viewBox="0 0 16 16" class="bi bi-search"><path d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"></path></svg></span> <input type="text" placeholder="Search by infant formula name..." aria-label="Baby Formula Name" class="form-control"> <button type="button" id="button-addon1" class="btn btn-dark round">Search</button></form>
  </section>
</div>"
`;
